edx_solutions_api_integration/users/tests.py

```diff
@@ -2189,3 +2189,25 @@ def test_mark_notification_as_read(self):
 
         # then verify unread count, which should be 0
         self.assertEqual(get_notifications_count_for_user(user_id, filters={'read': False}), 0)
+
+    @mock.patch("edx_solutions_api_integration.users.views.module_render.get_module_for_descriptor")
+    def test_user_courses_detail_get_undefined_course_module(self, mock_get_module_for_descriptor):
+        course_id = unicode(self.course.id)
+
+        # Enroll test user in test course
+        test_uri = '{}/{}/courses'.format(self.users_base_uri, self.user.id)
+        data = {'course_id': course_id}
+
+        response = self.do_post(test_uri, data)
+        self.assertEqual(response.status_code, 201)
+
+        # Get user course details when course_module is None
+        mock_get_module_for_descriptor.return_value = None
```

Comment:
I think we should not mock `get_module_for_descriptor`. we should test with actual scenario which is getting course details of a user without completing any modules.